src/prefect/tasks/mixpanel/mixpanel_tasks.py

```diff
@@ -0,0 +1,209 @@
+
+        Returns:
+            - if `parse_response` is False, then returns a `str` response pulled
+                from the Export API, (which is basically a JSONL string).
+            - if `parse_response` is True and `group_events` is True, then returns a `dict` where
+                each key contains homogeneous events.
+            - if `parse_response` is True and `group_events` is False, then returns
+                a `list` of JSON objects obtained by parsing the response.
+
+        Raises:
+            - `ValueError` if both `api_secret` and `api_secret_env_var` are missing.
+            - `ValueError` if `api_secret` is missing and `api_secret_env_var` is not found.
+            - `prefect.engine.signals.FAIL` if the Mixpanel API returns an error.
+
+        """
+        if not api_secret and not api_secret_env_var:
+            raise ValueError("Missing both `api_secret` and `api_secret_env_var`.")
+        elif not api_secret and api_secret_env_var not in os.environ:
+            raise ValueError("Missing `api_secret` and `api_secret_env_var` not found.")
+
+        secret = None
+        if api_secret:
+            self.logger.debug("Got secret from `api_secret`")
+            secret = api_secret
+        else:
+            self.logger.debug(
+                "Got secret from env var passed from `api_secret_env_var`"
+            )
+            secret = os.environ[api_secret_env_var]
+
+        params = {"from_date": from_date, "to_date": to_date}
+
+        if limit:
+            params["limit"] = limit
+
+        if event:
+            params["event"] = json.dumps([event] if isinstance(event, str) else event)
+
+        if where:
+            params["where"] = where
+
+        url = "https://{server}.mixpanel.com/api/2.0/export".format(
+            server="data-eu" if use_eu_server else "data"
+        )
+
+        response = requests.get(
+            url=url,
+            auth=HTTPBasicAuth(secret, ""),
+            headers={"Accept": "application/json"},
+            params=params,
+        )
+
+        if response.status_code != 200:
+            msg = f"""
+            Mixpanel export API error.
+            Status code: {response.status_code}
+            Reason: {response.reason}
+            Text: {response.text}
+            """
+            raise FAIL(message=msg)
+
+        events = response.text
+
+        ret = events if events else None
+
+        if ret:
+
+            if parse_response:
+                received_events = [json.loads(event) for event in events.splitlines()]
+
+                if group_events:
+                    grouped_events = defaultdict(list)
+                    for received_event in received_events:
+                        grouped_events[received_event["event"]].append(
+                            received_event["properties"]
+                        )
+                    return dict(grouped_events)
+
+                return received_events
+
+        return ret
```

Comment:
Here's a suggestion to remove one layer of conditional nesting
```suggestion
        if not events:
            return None
        elif parse_response:
            received_events = [json.loads(event) for event in events.splitlines()]
            if group_events:
                grouped_events = defaultdict(list)
                    for received_event in received_events:
                        grouped_events[received_event["event"]].append(
                            received_event["properties"]
                        )
                    return dict(grouped_events)
             return received_events
        else:
            return events
```